src/components/data/data-table-provenance.js

```diff
@@ -38,3 +38,63 @@ export const DataTableColumnProvenance = ({ workspace, column, provenance }) =>
     ])
   ])
 }
+
+export const FileProvenance = ({ workspace, fileUrl }) => {
+  const { workspace: { namespace, name } } = workspace
+  const { fileProvenance, loading, error } = useFileProvenance(workspace, fileUrl)
+
+  return Utils.cond(
+    [loading, () => 'Loading provenance...'],
+    [error, () => 'Unable to load provenance information.'],
+    () => Utils.switchCase(fileProvenance.type,
+      [fileProvenanceTypes.externalFile, () => 'Unknown. Provenance information is only available for files in the workspace bucket.'],
+      [fileProvenanceTypes.unknown, () => 'Unknown. This file does not appear to be associated with a submission.'],
+      [fileProvenanceTypes.maybeSubmission, () => span([
+        'Unknown. This file may be associated with submission ',
+        h(Link, {
+          href: Nav.getLink('workspace-submission-details', {
+            namespace, name,
+            submissionId: fileProvenance.submissionId
+          })
+        }, [fileProvenance.submissionId]),
+        ', but it was not found in workflow outputs.'
+      ])],
+      [fileProvenanceTypes.workflowOutput, () => span([
+        'This file is an output of workflow ',
```

Comment:
These links are a pretty bad user experience for screenreaders. The link text itself is a UUID that is meaningless, and the context around it isn't read out if you are only navigating links/focusable components.

I would suggest adding an aria-label to the links that describe what they are, and I don't think there is any need to actually read out the ID (which will be the behavior-- the label will be read instead of the ID). So for instance, here it would be "parent workflow", above it would be "possible parent submission", etc.

Reply:
Added labels in https://github.com/DataBiosphere/terra-ui/pull/3353/commits/f21e9e6119eb4eb3434f4c751cd078e2c3a31c56.